### PR TITLE
Quickswap long running test 

### PR DESCRIPTION
**Pull request by thala321**
We need to clean up a few things before running the test
```
docker system prune -a
make kill-all-tmux
```

Steps to run the test in Vu's computer
```
# Download code and build them manually
ssh massbit@6.tcp.ngrok.io -p 18268
cd work/massbitprotocol
git pull
git checkout quickswap-long-running
cargo build
cd store/postgres
cargo build
cd ../..

# Install prerequisites
make init-code-compiler

# Start docker min
docker-compose -f docker-compose.min.yml up -d

# Start our long running test
make test-long-running-quickswap
```

Other scripts:
- To manually reload schema
```
curl --location --request POST 'http://localhost:8080/v1/query' \
--header 'Content-Type: application/json' \
--data-raw '{
    "type" : "reload_metadata",
    "args": {
        "reload_remote_schemas": true
    }
}'
```
- To check if the quickswap test is still running
```
cd e2e-test/production
robot --variable  SSH_HOST:6.tcp.ngrok.io --variable SSH_PORT:18268 --variable SSH_USERNAME:massbit --variable SSH_PASSWORD:[ADD_PASSWORD_HERE] production.robot
```

### CODE REVIEW

**Comment by thala321** [comment]
E2E tests should have 4 categories:

Basic: test block, event, transaction
Advanced: compound type,...
Chain: get metrics from the chain (trading volume, active address from BSC/Ethereum/Polygon)
Contract: get metrics from the contract (pancakeswap, uniswap)
The reason we break into many small categories is that we have too many tests now, and sometimes we don't want to run them all. By separating like this, we can define the tests that we want to run.

Currently, we have:

BSC has 1 contract test (quickswap)
Ethereum has 3 basic tests (block, transaction, event)
Polygon has 1 contract test (quickswap), 1 chain test (transaction metrics)
Solana has 3 basic tests (block, transaction, log messages), 2 advanced tests (serum, five tables to test compound type)
Substrate has 3 basic tests (block, transaction, event)
Cardano none tests yet.

File: chain-reader/chain-reader/src/ethereum_chain.rs [hunk 1]
```diff
@@ -256,65 +272,59 @@ async fn get_block(
...
-    // Get version
-    let version = web3
+    let version = WEB3
         .net()
         .version()
         .wait()
         .unwrap_or("Cannot get version".to_string());
 
     let sem = Arc::new(Semaphore::new(BLOCK_BATCH_SIZE as usize));
+    let mut got_block_number = match start_block {
```
Comment (vuviettai) [inline chain-reader/chain-reader/src/ethereum_chain.rs:304]: Todo:Remove this conversion

File: chain-reader/chain-reader/src/grpc_stream.rs [hunk 2]
```diff
@@ -43,29 +46,47 @@ impl Streamout for StreamService {
         info!("Request = {:?}", request);
         let chain_type: ChainType = ChainType::from_i32(request.get_ref().chain_type).unwrap();
         let network: NetworkType = request.get_ref().network.clone();
+        let (tx, rx) = mpsc::channel(QUEUE_BUFFER);
+        match chain_type {
```
Comment (vuviettai) [inline chain-reader/chain-reader/src/grpc_stream.rs:50]: Todo: [WIP] -Refactor Solana/Substrate chain reader like ethereum

**Review by vuviettai — APPROVED** [review]
LGTM

**Pull request merged**